# Fraudar_demo

In [1]:
import sys
import spartan as st

Using backend cpu


Using backend cpu

In [7]:
# load graph data
tensor = st.loadTensor(path = "inputData/plain_graph_small.zip", col_types = [int, int], sep=' ')

stensor = tensor.toSTensor(hasvalue=False)
import pandas as pd
df = pd.DataFrame(tensor.data, columns=["user_id", "item_id"])

# 保存成 CSV（无索引、逗号分隔，标准格式）
df.to_csv("plain_graph_small.csv", index=False, encoding="utf-8")


In [3]:
# create a anomaly detection model
fd = st.Fraudar(stensor)

In [4]:
# run the model
fd.run(k=2, maxsize=[100, 100])

finished computing weight matrix
finished setting up greedy
InitAveScore: 0.2757912596294998 with shape (500, 497)
finished setting deltas
finished building min trees
Fraudar iter 0 finished.
n_row 20 n_col: 20
score obtained is  3.0490293632985193
finished computing weight matrix
finished setting up greedy
InitAveScore: 0.15816224053061087 with shape (500, 497)
finished setting deltas
finished building min trees
Fraudar iter 1 finished.
n_row 4 n_col: 12
score obtained is  0.4185829699133876


finished computing weight matrix  
finished setting up greedy  
finished setting deltas  
finished building min trees  
Fraudar iter 0 finished.
score obtained is  3.0490293632985193

finished computing weight matrix
finished setting up greedy
finished setting deltas
finished building min trees
Fraudar iter 1 finished.
score obtained is  0.4185829699133876